velox/functions/prestosql/types/IPAddressType.cpp

```diff
@@ -0,0 +1,216 @@
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto intAddr = ipaddresses->valueAt(row);
+      folly::ByteArray16 addrBytes;
+      std::string s;
+      memcpy(&addrBytes, &intAddr, kIPAddressBytes);
+
+      bigEndianByteArray(addrBytes);
+      folly::IPAddressV6 v6Addr(addrBytes);
+
+      if (v6Addr.isIPv4Mapped()) {
+        s = v6Addr.createIPv4().str();
+      } else {
+        s = v6Addr.str();
+      }
+
+      exec::StringWriter<false> result(flatResult, row);
+      result.append(s);
+      result.finalize();
+    });
+  }
+
+  static void castFromString(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<int128_t>>();
+    const auto* ipAddressStrings = input.as<SimpleVector<StringView>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto ipAddressString = ipAddressStrings->valueAt(row);
+      int128_t intAddr;
+      folly::IPAddress addr(ipAddressString);
+
+      auto addrBytes = folly::IPAddress::createIPv6(addr).toByteArray();
+
+      bigEndianByteArray(addrBytes);
+      memcpy(&intAddr, &addrBytes, kIPAddressBytes);
+
+      flatResult->set(row, intAddr);
+    });
+  }
+
+  static void castToVarbinary(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<StringView>>();
+    const auto* ipaddresses = input.as<SimpleVector<int128_t>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      const auto intAddr = ipaddresses->valueAt(row);
+      folly::ByteArray16 addrBytes;
+      memcpy(&addrBytes, &intAddr, kIPAddressBytes);
+      bigEndianByteArray(addrBytes);
+
+      exec::StringWriter<false> result(flatResult, row);
+      result.resize(kIPAddressBytes);
+      memcpy(result.data(), &addrBytes, kIPAddressBytes);
+      result.finalize();
+    });
+  }
+
+  static void castFromVarbinary(
+      const BaseVector& input,
+      exec::EvalCtx& context,
+      const SelectivityVector& rows,
+      BaseVector& result) {
+    auto* flatResult = result.as<FlatVector<int128_t>>();
+    const auto* ipAddressBinaries = input.as<SimpleVector<StringView>>();
+
+    context.applyToSelectedNoThrow(rows, [&](auto row) {
+      int128_t intAddr;
+      folly::ByteArray16 addrBytes = {};
+      const auto ipAddressBinary = ipAddressBinaries->valueAt(row);
+
+      if (ipAddressBinary.size() == kIPV4AddressBytes) {
+        addrBytes[kIPV4ToV6FFIndex] = 0xFF;
+        addrBytes[kIPV4ToV6FFIndex + 1] = 0xFF;
+        memcpy(
+            &addrBytes[kIPV4ToV6Index],
+            ipAddressBinary.data(),
+            kIPV4AddressBytes);
+      } else if (ipAddressBinary.size() == kIPAddressBytes) {
+        memcpy(&addrBytes, ipAddressBinary.data(), kIPAddressBytes);
+      } else {
+        context.setStatus(
```

Comment:
let's use threadSkipErrorDetails to avoid creating error message if it is not needed

Reply:
Done.